### PR TITLE
Fix/ pythran extension side effect improvement

### PR DESCRIPTION
**Pull request by ashwinvis**
Improves #936

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
Thanks! I'll check that in the forthcoming days.

File: pythran/config.py [hunk 1]
```diff
@@ -112,8 +112,18 @@ def parse_define(define):
...
+    """Get compiler to use for C++ to binary process. The precedence for
+    choosing the compiler is as follows::
+
+      1. `CXX` environment variable
+      2. User configuration (~/.pythranrc)
+      3. Default to `c++`
+
+    """
+    cfg_cxx = str(cfg.get('compiler', 'CXX'))
+    if cfg_cxx == '':
```
Comment (serge-sans-paille) [inline pythran/config.py]: Woul be more pythonic to write ``if not cfg_cxx:``

File: pythran/dist.py [hunk 2]
```diff
@@ -7,44 +7,40 @@
 
 import os.path
 import os
+from warnings import warn
```
Comment (serge-sans-paille) [inline pythran/dist.py]: There's already a logging mechanism in pythran, with a warning level, please use it :-)

File: pythran/dist.py [hunk 3]
```diff
@@ -7,44 +7,40 @@
...
+    """
+    def build_extension(self, ext):
+        prev = {'preprocessor': None,
+                'compiler_cxx': None,
+                'compiler_so': None,
+                'compiler': None,
+                'linker_exe': None,
+                'linker_so': None}
+
+        if hasattr(ext, 'cxx'):
```
Comment (serge-sans-paille) [inline pythran/dist.py]: I know it's my original code, but maybe a comment here to state we're saving the environment?

File: pythran/dist.py [hunk 4]
```diff
@@ -72,6 +68,7 @@ def __init__(self, name, sources, *args, **kwargs):
 
         import pythran.toolchain as tc
 
+
```
Comment (serge-sans-paille) [inline pythran/dist.py]: noise :-)

**Comment by serge-sans-paille** [comment]
Only problem with travis: as there's a new warning, one need to accept the warning in the notebook validation, see 
https://travis-ci.com/serge-sans-paille/pythran/jobs/129607785

I can deal with it if you dont wan't to bother with that.

**Comment by ashwinvis** [comment]
All done, except for the notebook validation. I guess you can take it from here, @serge-sans-paille.

**Comment by ashwinvis** [comment]
Obsoleted by #941

**Pull request closed without merging**